decoder/rubix.go

```diff
@@ -0,0 +1,413 @@
+	case *int64:
+		fmt.Println(*v)
+	case *float32:
+		fmt.Println(*v)
+	default:
+		fmt.Println("Unsupported type:", reflect.TypeOf(value))
+	}
+}
+
+func decode(serialData *SerialData) {
+	header := getHeader(serialData, nil)
+	switch header {
+	case MDK_CHAR:
+		var value byte
+		decodeAndPrint(serialData, header, &value)
+	case MDK_UINT_8:
+		var value uint8
+		decodeAndPrint(serialData, header, &value)
+	case MDK_UINT_16:
+		var value uint16
+		decodeAndPrint(serialData, header, &value)
+	case MDK_UINT_32:
+		var value uint32
+		decodeAndPrint(serialData, header, &value)
+	case MDK_UINT_64:
+		var value uint64
+		decodeAndPrint(serialData, header, &value)
+	case MDK_INT_8:
+		var value int8
+		decodeAndPrint(serialData, header, &value)
+	case MDK_INT_16:
+		var value int16
+		decodeAndPrint(serialData, header, &value)
+	case MDK_INT_32:
+		var value int32
+		decodeAndPrint(serialData, header, &value)
+	case MDK_INT_64:
+		var value int64
+		decodeAndPrint(serialData, header, &value)
+	default:
+		var value float32
+		decodeAndPrint(serialData, header, &value)
+	}
+}
+
+var serialMap = map[int]MetaData{
+	MDK_TEMP:             {FIXEDPOINT, -45, 120, 2, 0},
+	MDK_RH:               {FIXEDPOINT, 0, 100, 2, 0},
+	MDK_LUX:              {FIXEDPOINT, 0, 65534, 0, 0},
+	MDK_MOVEMENT:         {FIXEDPOINT, 0, 1, 0, 0},
+	MDK_COUNTER:          {FIXEDPOINT, 0, 1048576, 0, 0},
+	MDK_DIGITAL:          {FIXEDPOINT, 0, 1, 0, 0},
+	MDK_VOLTAGE_0_10:     {FIXEDPOINT, 0, 10, 2, 0},
+	MDK_MILLIAMPS_4_20:   {FIXEDPOINT, 4, 20, 2, 0},
+	MDK_OHM:              {FIXEDPOINT, 0, 1048576, 0, 0},
+	MDK_CO2:              {FIXEDPOINT, 0, 400, 0, 0},
+	MDK_BATTERY_VOLTAGE:  {FIXEDPOINT, 0, 6, 1, 0},
+	MDK_PUSH_FREQUENCY:   {FIXEDPOINT, 0, 2000, 0, 0},
+	MDK_RAW:              {FIXEDPOINT, 0, 1, 3, 0},
+	MDK_UO:               {FIXEDPOINT, 0, 1, 3, 0},
+	MDK_UI:               {FIXEDPOINT, 0, 1, 3, 0},
+	MDK_DO:               {FIXEDPOINT, 0, 1, 0, 0},
+	MDK_DI:               {FIXEDPOINT, 0, 1, 0, 0},
+	MDK_FIRMWARE_VERSION: {FIXEDPOINT, 0, 255, 0, 0},
+	MDK_HARDWARE_VERSION: {FIXEDPOINT, 0, 255, 0, 0},
+	MDK_UINT_8:           {DATAPOINT, 0, 0, 0, 1},
+	MDK_INT_8:            {DATAPOINT, 0, 0, 0, 1},
+	MDK_UINT_16:          {DATAPOINT, 0, 0, 0, 2},
+	MDK_INT_16:           {DATAPOINT, 0, 0, 0, 2},
+	MDK_UINT_32:          {DATAPOINT, 0, 0, 0, 4},
+	MDK_INT_32:           {DATAPOINT, 0, 0, 0, 4},
+	MDK_UINT_64:          {DATAPOINT, 0, 0, 0, 8},
+	MDK_INT_64:           {DATAPOINT, 0, 0, 0, 8},
+	MDK_BOOL:             {DATAPOINT, 0, 0, 0, 0},
+	MDK_CHAR:             {DATAPOINT, 0, 0, 0, 1},
+	MDK_FLOAT:            {DATAPOINT, 0, 0, 0, 4},
+	MDK_DOUBLE:           {DATAPOINT, 0, 0, 0, 8},
+}
+
+func DecodeRubix(data string, _ *LoRaDeviceDescription) (*CommonValues, interface{}) {
+	bytes, err := hex.DecodeString(data)
+	if err != nil {
+		log.Errorf("error decoding rubix hex data")
+		return nil, nil
+	}
+	sd := NewSerialDataWithBuffer(bytes)
+
+	// TODO: implement decode loop
```

Comment:
@lmtan91 See JS example of decode loop implementation here: https://github.com/NubeIO/rubix-data-encoding/blob/master/js/decode-ES5.js#L194-L219